www/src/components/shell/Shell.tsx

```diff
@@ -71,28 +68,36 @@ function Shell({ shell }: any) {
       // }
     })
     chan.join()
+    setChannel(chan)
+
+    return () => {
+      socket.off([ref])
+      chan.leave()
+    }
+  }, [shell, xterm, restart])
```

Comment:
Wanna add that missing `ref` to dependency array?

Reply:
I actually think that should be removed entirely.  Kind of surprised it didn't barf at there being no variable named ref in scope